aten/src/ATen/native/Memory.cpp

```diff
@@ -19,15 +25,28 @@ int64_t _debug_has_internal_overlap(const Tensor& self) {
   return static_cast<int64_t>(at::has_internal_overlap(self));
 }
 
-// Technically, we could force backends to explicitly say "no, we don't support
-// pinned memory, always return false", but this makes life a little easier when
-// you haven't loaded the backend extension at all (which can happen, e.g., on a
-// CPU build of PyTorch and you try to check if something is CUDA pinned)
-bool is_pinned_default(const Tensor& self, std::optional<Device> device) {
-  return false;
```

Comment:
As you mentioned, we definitely want to preserve this behavior.
In this case, I think we need to properly handle the case where there is no accelerator or the passed device is not an accelerator and always return "false" for these.
That should preserve the old behavior for non-accelerator devices right?

Reply:
@albanD Yes, updated.